test/testchecksumvalidator.cpp

```diff
@@ -84,10 +86,11 @@ using namespace OCC::Utility;
         QFileInfo fi(file);
         QVERIFY(fi.exists());
 
-        QFile fileDevice(file);
-        fileDevice.open(QIODevice::ReadOnly);
-        QByteArray sum = calcMd5(&fileDevice);
-        fileDevice.close();
+        QSharedPointer<QFile> fileDevice(new QFile(file));
```

Comment:
```suggestion
        auto fileDevice(QSharedPointer<QFile>::create(file));
```
please do the same in every other places building instances of `QSharedPointer`